src/analyses/libraryFunctions.ml

```diff
@@ -9,7 +9,7 @@ type categories = [
   | `Malloc       of exp
   | `Calloc       of exp * exp
   | `Realloc      of exp * exp
-  | `Assert       of exp
+  | `Assert       of exp * bool * bool (* exp, should_warn?, change? *) (* TODO: change to non-poly variants to inline record can be used *)
```

Comment:
Would it not make sense to split this into some sort of `GoblintAssert` that is e.g. the result of classify `__goblint_check`, and keep this completely separate from `Assert` and use `Assert` only for actual C asserts?

Then one could have a separate anlaysis for these "regression test" `__goblint_check` (which is the analysis you started in this PR). For actual `assert` the default behavior would always be `change` and there might be a separate analysis that also does the warnings for C-level `asserts` that do not hold.

Reply:
This would also make it easy to warn for programs that contain `__goblint_check` but are run without this "regression assert" analysis, whereas the behavior changing a lot depending on whether the assert analysis is active or not might be confusing for some.

Reply:
Splitting `assert` from `__goblint_*` might be a bit cleaner indeed, but I'll postpone those changes also to after GobCon to have decided what's our plan with all these things.

Splitting this analysis itself also in two (C `assert()` analysis and our `__goblint_*` assert analysis) might be too much hair splitting. We have the `dbg.regression` option, so the same assert analysis could also warn if that's disabled but `__goblint_*` is called.